docs/recipes/type-checking/tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "noEmit": true,
+    "target": "esnext",
+    "module": "esnext",
```

Comment:
I don't believe it technically affects anything because there's no emit, but `module` and `moduleResolution` should both be `'NodeNext'` if one of them is (I think that might be an error in the future?)